qp_qiime2/__init__.py

```diff
@@ -76,234 +86,191 @@
+                to_delete.append(pname)
+            elif predicate == ['Sequence']:
+                req_params[element.description] = ('artifact', ['BIOM'])
+                to_delete.append(pname)
+            else:
+                add_method = False
+        elif etype == 'TaxonomicClassifier':
+            default = qp_qiime2_dbs[0]
+            qp_qiime2_dbs = ', '.join('"%s"' % db for db in qp_qiime2_dbs)
 
-            for pname, element in parameters.items():
-                tqt, predicate = get_qiime2_type_name_and_predicate(element)
-                # there is a new primitive and we should raise an error
-                if tqt not in PRIMITIVE_TYPES:
-                    raise ValueError(
-                        'There is a new type: %s, in %s %s (%s)' % (
-                            tqt, q2plugin.name, m.id, pname))
-
-                # predicate are the options for each parameter, note that it
-                # can be a Choice/List or a Range (for Int/Floats). We ignore
-                # numeric because they are ranges and we don't support ranges
-                # in Qiita.
-                # Note, the correct way to retrieve the latter is:
-                # p.start, p.end, p.inclusive_start, p.inclusive_end
-                # but for simplicity we will only retrive the
-                data_type = PRIMITIVE_TYPES[tqt]
-                default = element.default
-                if (predicate is not None and PRIMITIVE_TYPES[tqt] not in (
-                                              'float', 'integer')):
-                    vals = predicate.to_ast()['choices']
-                    data_type = 'choice:%s' % dumps(vals)
-                    default = vals[0]
+            ename = '%s (%s)' % (element.description, pname)
+            req_params[ename] = ('choice:[%s]' % qp_qiime2_dbs, default)
+            req_params['qp-hide-param' + ename] = ('string', pname)
+        else:
+            ename = element.description
+            # this is an odd one, first encountered:
+            # feature-classifier fit-classifier-naive-bayes
+            if ename == element.NOVALUE:
+                add_method = False
+            else:
+                req_params[ename] = ('artifact', [etype])
+                req_params['qp-hide-param' + ename] = ('string', pname)
 
-                qname = q2plugin.name
-                mid = m.id
-                # if we are in the diversity plugin, the method starts with
-                # alpha/beta, and the parameter is 'metric', we might want
-                # to replace the technical names for user friendly ones
-                value_pair = (mid, pname)
-                if qname == 'diversity' and value_pair in RENAME_COMMANDS:
-                    # converting to list to the serialize doesn't complaint
-                    vals = list(RENAME_COMMANDS[value_pair])
-                    data_type = 'choice:%s' % dumps(vals)
-                    default = vals[0]
+    if add_method:
```

Comment:
This is also new, in the past, we will still run the code within the if, even if `add_method ` was already set as `False` so now skipping.